tests/unit/utils/notification/test_v0.py

```diff
@@ -0,0 +1,345 @@
+            get_html_mock.assert_called_with(notification_type=test_notification_type,
+                                             options=test_options,
+                                             sender=test_sender)
+
+    def test_get_notification_html_no_resource_url(self) -> None:
+        test_notification_type = 'added'
+        test_sender = 'test@test.com'
+        test_options = {'resource_name': 'testtable'}
+
+        self.assertRaises(Exception,
+                          get_notification_html,
+                          notification_type=test_notification_type,
+                          options=test_options,
+                          sender=test_sender)
+
+    def test_get_notification_html_no_resource_name(self) -> None:
+        test_notification_type = 'added'
+        test_sender = 'test@test.com'
+        test_options = {'resource_url': 'testUrl'}
+
+        self.assertRaises(Exception,
+                          get_notification_html,
+                          notification_type=test_notification_type,
+                          options=test_options,
+                          sender=test_sender)
+
+    def test_get_notification_html_unsupported_type(self) -> None:
+        test_notification_type = 'invalid_type'
+        test_sender = 'test@test.com'
+        test_options = {'resource_name': 'testtable', 'resource_url': 'testUrl'}
+
+        self.assertRaises(Exception,
+                          get_notification_html,
+                          notification_type=test_notification_type,
+                          options=test_options,
+                          sender=test_sender)
+
+    def test_get_notification_html_added_success(self) -> None:
+        test_notification_type = 'added'
+        test_sender = 'test@test.com'
+        test_options = {'resource_name': 'testtable', 'resource_url': 'testUrl'}
+
+        html = get_notification_html(notification_type=test_notification_type,
+                                     options=test_options,
+                                     sender=test_sender)
+        expectedHTML = ('Hello,<br/><br/>You have been added to the owners list of the <a href="testUrl">testtable</a>'
+                        ' dataset by test@test.com.<br/><br/>What is expected of you?<br/>As an owner, you take an '
+                        'important part in making sure that the datasets you own can be used as swiftly as possible '
+                        'across the company.<br/>Make sure the metadata is correct and up to date.<br/><br/>If you '
+                        'think you are not the best person to own this dataset and know someone who might be, please '
+                        'contact this person and ask them if they want to replace you. It is important that we keep '
+                        'multiple owners for each dataset to ensure continuity.<br/><br/>Thanks,<br/>Amundsen Team')
```

Comment:
I wonder if there's a better way to test the HTML templates other than comparing the full text for an exact match. Can't think of any right now.

Reply:
Yep...going to have to change the test if you add even a new character. 

Ideally though I think it involves exporting these hardcoded strings as constants, creating separate methods for getting every  piece of the email (e.g. `_generate_greeting()`, `_generate_salutation()`, `_generate_comment()`) and then testing the the top level method outputs the concatenation of all those helper methods, and that the helper methods themselves output `CONSTANT_FOR_HARCDODED_TEXT.format(...test parameters..)`.  

I'm up for doing that as  a cleanup after I get past the blockers for prod and merging this down.